htdocs/core/modules/export/export_csv.modules.php

```diff
@@ -279,14 +288,22 @@ public function write_record($array_selected_sorted, $objp, $outputlangs, $array
 			$newvalue = $this->csvClean($newvalue, $outputlangs->charset_output);
 
 			if (preg_match('/^Select:/i', $typefield) && $typefield = substr($typefield, 7)) {
-				$array = json_decode($typefield, true);
-				$array = $array['options'];
-				$newvalue = $array[$newvalue];
+				$array = jsonOrUnserialize($typefield);
```

Comment:
jsonOrUnserialize has been introduct to start migration from storing data using "serialize" (not secured) into storing data using "json". Here old code is already using json, so jsonOrUnserialize seems not necessary, we shoul be able to keep json_decode (infeature we want only json_decode ans json_encode instead of serialize, unserialize).
What happen if we keep 
`$array = json_decode($typefield, true);`